Feature/books

immortaldogg opened the pull request:
Add CRUD for books + association between books and categories
- Can create/get/update/delete books
- Validate name/description length and category_id existence before creating a book
- Delete a category will lead to deleting all books associated with that category

immortaldogg committed:
add category crud operation

- Add domain model "category"
- Add endpoints, decode, service for "category"
- Add middleware validation + error codes for "category"
- Add route for "category"
- Add sql command for creating "categories" table

immortaldogg committed:
Add CRUD for books + association between books and categories

- Can create/get/update/delete books
- Validate name/description length and category_id existence before creating a book
- Delete a category will lead to deleting all books associated with that category

fuatto commented on service/book/mw_validation.go:
```diff
@@ -0,0 +1,71 @@
+	return mw.Service.Create(ctx, book)
+}
+
+// func (mw validationMiddleware) FindAll(ctx context.Context, nameToFind string) ([]domain.Book, error) {
```
Either remove not using code or uncomment it

immortaldogg replied:
Resolve in newest commit

fuatto commented on service/book/error.go:
```diff
@@ -0,0 +1,92 @@
+func (errDescriptionIsRequired) Error() string {
+	return "book description is required"
+}
+
```
Should have implement StatusCode()

immortaldogg replied:
Resolve in newest commit

immortaldogg committed:
Add missing statuscode and validations
- Add status code to book error
- Uncomment and add validation to Update book API
